web-app/src/components/Legend/container.js

```diff
@@ -12,130 +12,91 @@ const LegendContainer = Wrapped =>
-        if (a.displayName > b.displayName) return 1;
-        return 0;
-      });
-    };
+    storeLegendKeysToState = () => {
+      const statusKeys = Object.keys(holidayStatus);
+      const statusList = [];
+      const typeList = [];
 
-    createEmployeeObject = hol => {
-      let { employeeId, forename, surname } = hol.employee;
-      return {
-        value: employeeId,
-        displayValue: `${forename} ${surname}`,
-      };
-    };
-
-    removeDuplicateEmployees = employees => {
-      return employees.reduce((unique, o) => {
-        if (
-          !unique.some(
-            obj => obj.value === o.value && obj.displayValue === o.displayValue
-          )
-        ) {
-          unique.push(o);
+      for (const index of statusKeys.keys()) {
+        const keyObj = {
+          eventId: index + 1,
+          key: index + 1,
+          active: false,
+        };
+        // temp if statement
+        if (index < 4) {
+          statusList.push(keyObj);
+        } else {
+          typeList.push(keyObj);
         }
-        return unique;
-      }, []);
-    };
+      }
 
-    createEmployeesList = takenHolidays => {
-      let employees = takenHolidays.filter(hol => hol.employee).map(hol => {
-        return this.createEmployeeObject(hol);
+      this.setState({
+        statusList: statusList,
+        typesList: typeList,
       });
-      employees = this.removeDuplicateEmployees(employees);
-      employees.unshift({ value: -1, displayValue: 'All' });
-      this.setState({ employees: this.sortEmployeeList(employees) });
     };
 
-    setKeyActiveState = eventStatusId => {
-      let updatedList = [...this.state.eventsKeys];
+    storeEmployeeListToState = () => {
+      const { takenHolidays } = this.props;
+      const employeeList = takenHolidays
+        .filter(hol => hol.employee)
+        .map(hol => hol.employee)
+        .filter((employee, pos, arr) => {
+          return (
+            arr
+              .map(mapObj => mapObj.employeeId)
+              .indexOf(employee.employeeId) === pos
+          );
+        });
+      this.setState({ employeeList });
+    };
+
+    setLegendKeyActiveState = (eventId, event) => {
+      let updatedList;
+      let updatedProp;
+      if (event === 'status') {
+        updatedList = [...this.state.statusList];
+        updatedProp = 'statusList';
+      } else {
+        updatedList = [...this.state.typesList];
+        updatedProp = 'typesList';
+      }
       for (let key of updatedList) {
-        if (key.eventStatusId === eventStatusId) {
+        if (key.eventId === eventId) {
           key.active = !key.active;
         }
       }
-      this.setState({
-        selectedEventStatusID: eventStatusId,
-      });
-      this.props.updateCalendarEvents(eventStatusId);
+      this.setState({ [updatedProp]: updatedList });
+      this.props.updateCalendarEvents(eventId, event);
     };
 
-    handleFormStatus(name, value) {
+    onFilterUserChange(_, value) {
```

Comment:
The the name param is not required for this function even though it is being passed in. saw 
 underscore being used as a param for componentDidUpdate in pages -> dashboard -> container and thought i'd reuse. Is this bad practice?

Reply:
@paul-matchett If `name` is never being used in the function any time it's being called there is no point of it being a parameter. The usage in componentDidMount is because we don't want to use `prevProps` but we need access to the second parameter `prevState`, however, there are obvious use cases where prevProps is required. 

TypeScript sees this as expected behaviour (according to VSCode errors) and it's practice in other languages.